src/backend/tools/py.CLI/caliopen_cli/commands/resync_index.py

```diff
@@ -24,21 +24,33 @@ def resync_index(user_name, **kwargs):
     from caliopen_main.message.store import Message
     from caliopen_main.message.objects import Message as MessageObject
 
-    user = User.by_name(user_name)
+    if 'user_name' in kwargs and kwargs['user_name']:
+        user = User.by_name(kwargs['user_name'])
+    elif 'user_id' in kwargs and kwargs['user_id']:
+        user = User.get(kwargs['user_id'])
+    else:
+        print('Need user_name or user_id parameter')
+        sys.exit(1)
 
     mapping_key = 'elasticsearch.mappings_version'
-    index_version = Configuration('global').get(mapping_key)
-    index = '{}_{}'.format(user.user_id, index_version)
+    current_version = Configuration('global').get(mapping_key)
+    new_index = '{}_{}'.format(user.user_id, current_version)
+
+    if 'version' in kwargs and kwargs['version']:
+        old_version = kwargs['version']
+        old_index = '{}_{}'.format(user.user_id, old_version)
+    else:
+        old_index = new_index
 
     es_url = Configuration('global').get('elasticsearch.url')
     es_client = Elasticsearch(es_url)
 
     # Delete current index
-    if not es_client.indices.exists([index]):
-        log.warn('Index not found for user %s' % user_name)
+    if not es_client.indices.exists([old_index]):
+        log.warn('Index %r not found for user %s' % (old_index, user.name))
         sys.exit(1)
```

Comment:
why are you exiting ? You may want to create an index if it is missing, won't you ?

Reply:
it's not a normal situation, may index is in a wrong version etc. I prefer to deal with that situation manually